### PR TITLE
Expanded test of species class and related builders/factory

### LINKED ISSUE #479: Expand testing of species
suggestion (testing): Add edge case tests for GasSpecies

Consider adding tests for edge cases, such as creating a GasSpecies with extreme values (very large or very small) for molar mass or concentration. Also, test the behavior when invalid inputs are provided.

def test_gas_species_builder_single_species():
    """Test building a single gas species with the GasSpeciesBuilder."""
    vapor_pressure_strategy = ConstantVaporPressureStrategy(vapor_pressure=101325)
    name = "Oxygen"
    molar_mass = 0.032  # kg/mol
    condensable = False
    concentration = 1.2  # kg/m^3

    gas_species = GasSpecies(
        name=name,
        molar_mass=molar_mass,
        vapor_pressure_strategy=vapor_pressure_strategy,
        condensable=condensable,
        concentration=concentration
    )

    assert gas_species.name == name
    assert gas_species.get_molar_mass() == molar_mass
    assert gas_species.get_condensable() is condensable
    assert gas_species.get_concentration() == concentration
    assert gas_species.get_pure_vapor_pressure(298) == 101325

def test_gas_species_builder_edge_cases():
    """Test edge cases for GasSpecies creation."""
    with pytest.raises(ValueError):
        GasSpecies(name="Invalid", molar_mass=-1, vapor_pressure_strategy=None, condensable=True, concentration=0)

    large_mass_species = GasSpecies(name="Heavy", molar_mass=1e10, vapor_pressure_strategy=None, condensable=False, concentration=1e-10)
    assert large_mass_species.get_molar_mass() == 1e10
    assert large_mass_species.get_concentration() == 1e-10

### PR DESCRIPTION
**Pull request by Gorkowski**
- expanded test of gas species class: adding value checks (with logging), and tests.
- updated docs in gas species, species builder, species factory , with example snips.
- added tests for species factory.

Fixes #479

## Summary by Sourcery

Enhance the GasSpecies class with logging and validation for invalid values, update related documentation with examples, and expand test coverage for GasSpecies and SpeciesFactory.

Enhancements:
- Add logging and validation for negative and non-positive values in GasSpecies class.

Documentation:
- Update documentation for GasSpecies, SpeciesBuilder, and SpeciesFactory with example snippets.

Tests:
- Expand tests for GasSpecies class to include validation of negative and zero values for molar mass and concentration.
- Add new tests for SpeciesFactory to ensure correct creation of gas species objects.

## Summary by Sourcery

Enhance the GasSpecies class with logging and validation for invalid values, update related documentation with examples, and expand test coverage for GasSpecies and SpeciesFactory.

Enhancements:
- Add logging and validation for negative and non-positive values in the GasSpecies class.

Documentation:
- Update documentation for GasSpecies, SpeciesBuilder, and SpeciesFactory with example snippets.

Tests:
- Expand tests for GasSpecies class to include validation of negative and zero values for molar mass and concentration.
- Add new tests for SpeciesFactory to ensure correct creation of gas species objects.

### CODE REVIEW

File: particula/gas/species.py [hunk 1]
```diff
@@ -58,11 +46,31 @@
         concentration: Union[float, NDArray[np.float64]] = 0.0,
     ) -> None:
         self.name = name
-        self.molar_mass = molar_mass
         self.pure_vapor_pressure_strategy = vapor_pressure_strategy
         self.condensable = condensable
+
+        if isinstance(concentration, np.ndarray):
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/gas/species.py:52]: made them the same

File: particula/gas/species.py [hunk 2]
```diff
@@ -58,11 +46,31 @@ def __init__(  # pylint: disable=too-many-positional-arguments
         concentration: Union[float, NDArray[np.float64]] = 0.0,
     ) -> None:
         self.name = name
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/gas/species.py:48]: done

File: particula/gas/species.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/gas/species.py:52]: fixed

File: particula/gas/species.py [hunk 3]
```diff
@@ -271,11 +298,45 @@
 
     def add_concentration(
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/gas/species.py:299]: done

File: particula/gas/tests/species_factory_test.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    def test_get_strategy_gas_species(self):
+        """Test getting a gas species strategy."""
+        parameters = {
+            "name": "Oxygen",
+            "molar_mass": 0.032,
+            "vapor_pressure_strategy": ConstantVaporPressureStrategy(50),
+            "condensable": False,
+            "concentration": 1.2,
+        }
+        gas_species = self.factory.get_strategy("gas_species", parameters)
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/gas/tests/species_factory_test.py:38]: done

File: particula/gas/tests/species_factory_test.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        self.assertIsInstance(gas_species, GasSpecies)
+
+    def test_get_strategy_preset_gas_species(self):
+        """Test getting a preset gas species strategy."""
+        parameters = {}
+        preset_gas_species = self.factory.get_strategy(
+            "preset_gas_species", parameters
+        )
+        self.assertIsInstance(preset_gas_species, GasSpecies)
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/gas/tests/species_factory_test.py:47]: done

**Comment by Gorkowski** [comment]
@sourcery-ai re-review

**Comment by Gorkowski** [comment]
@sourcery-ai review

File: particula/gas/species.py [hunk 6]
```diff
@@ -271,11 +282,50 @@ def get_saturation_concentration(
...
+                species.
+
+        Example:
+            ``` py title="Example usage of add_concentration"
+            gas_object.add_concentration(added_concentration=1e-10)
+            ```
+        """
+        self.set_concentration(self.concentration + added_concentration)
+
+    def set_concentration(
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/gas/species.py:299]: save that for a future refactoring

**Comment by Gorkowski** [comment]
@mahf708 , ok ready now
